controllers/flux/job.go

```diff
@@ -93,3 +94,16 @@ func (r *MiniClusterReconciler) getMiniClusterContainers(cluster *api.MiniCluste
 	}
 	return containers
 }
+
+// Function to return list of objects references for
+// imagePullSecrets
+func getImagePullSecrets(cluster *api.MiniCluster) []corev1.LocalObjectReference {
+	var pullSecrets []corev1.LocalObjectReference
+	var pullSecret corev1.LocalObjectReference
+	for _, secretName := range cluster.Spec.ImagePullSecrets {
```

Comment:
Yeah so I think we would be okay not having a list here, and setting just one secret. @milroy can you think of a case where there would be multiple containers or multiple secrets needed? I'm a flux noob so I might be missing something.

Reply:
Yeah, that's a fair point! I went ahead and applied those changes, we can always switch it back later if a use case arises

Reply:
Yes, there is a use case for this. The developer of AHA MoleS workflows and others would like to do inference with PyTorch running as a service. Flux and PyTorch would reside in separate containers within the same pod and use a shared process namespace or `localhost` to communicate.

Reply:
ah ok! So we will need a container-specific pull secret. @milroy do you have this example workflow we can use to develop? I think we want to develop this in parallel with a "I have two containers" use case.

Reply:
To be clear, I can't think of is a use case for multiple secrets

Reply:
If that example has multiple containers that should be sufficient to develop for that use case - I can just pretend that they need different secrets. :laughing: 

That workflow sounds cool too - I like PyTorch! :fire: